### DISCUSSION TITLE
Blank screen of the death with an ESP32-S2

### DISCUSSION THREAD

**Discussion by N0ury** [Bodmer/TFT_eSPI#2950] [Q&A - General]
I have succeeded in using a 2.8" ILI9341 LCD (with touch) with a lolin32D devkit board.

But I can't use this LCD with an ESP32-S2-SOLO-2-N4R2
N4R2 has 4MB (Quad SPI) Flash and 2 MB (Quad SPI) PSRAM
I use an ESP32-S2 module (not a devkit) via a custom PCB adapter.
I know my config works fine.

I try to use FSPI pins (same as in Setup70_ESP32_S2_ILI9341.h)
here's an extract from Espressif doc:
<img width="646" alt="Capture d’écran 2023-11-03 à 14 50 00" src="https://github.com/Bodmer/TFT_eSPI/assets/4224717/24017b4d-1a3e-4d84-b984-dd81670f2413">

here's my platformio.ini file:
```
[env]
platform = espressif32
board = esp32-s2-saola-1
framework = arduino
monitor_speed = 115200
upload_speed = 921600
board_upload.use_1200bps_touch = true
board_upload.wait_for_upload_port = true
board_build.flash_mode = qio
lib_deps =
        bodmer/TFT_eWidget@^0.0.6
        bodmer/TFT_eSPI@^2.5.33
build_flags =
        -D CORE_DEBUG_LEVEL=5
        -D ARDUINO_USB_CDC_ON_BOOT=1
        -D BOARD_HAS_PSRAM
        -mfix-esp32-psram-cache-issue
        -D CORE_DEBUG_LEVEL=5
        -D USER_SETUP_LOADED=1
        -D ESP32=1
        -D ILI9341_DRIVER=1
        -D TFT_WIDTH=240
        -D TFT_HEIGHT=320
        -D TFT_INVERSION_OFF=1
        -D TFT_MISO=37
        -D TFT_MOSI=35
        -D TFT_SCLK=36
        -D TFT_CS=34
        -D TFT_DC=14
        -D TFT_RST=21
        -D TFT_BL=13
;       -D TOUCH_CS=21
        -D TFT_BACKLIGHT_ON=HIGH
        -D SPI_FREQUENCY=40000000
        -D LOAD_GLCD=1
        -D LOAD_FONT2=1
        -D LOAD_FONT4=1
        -D LOAD_FONT6=1
        -D LOAD_FONT7=1
        -D LOAD_FONT8=1
        -D LOAD_GFXFF=1
        -D SMOOTH_FONT=1

[env:colour]
build_src_filter = +<colour.cpp>
```

I've tried using a FTDI board, and removing USB_CDC, it's the same.
I've trie to not use PSRAM, it's the same.
I have a blank screen which flickers sometimes.
colour.cpp app is Colour_Test.ino

Here's the output of:
```
[code]
TFT_eSPI ver = 2.5.33
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
Display driver = 9341
Display width  = 240
Display height = 320

MOSI    = GPIO 35
MISO    = GPIO 37
SCK     = GPIO 36
TFT_CS   = GPIO 34
TFT_DC   = GPIO 14
TFT_RST  = GPIO 21

TFT_BL           = GPIO 13
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 40.00
[/code]
```

I've been stuck for several days now. A little help would be great.

**Comment by N0ury** [comment]
Here are some more informations.
I know LCD works as it was usable with a Lolin32 board.
I know my ESP32S2-SOLO-2 works because I have used these SPI pins with a BMP280.
I have tried with previous MCU version (ESP32S2-SOLO, without the ending "-2"), and it's the same.
I have changed the pins to D10, D11, D12 and D13. It's the same.
I have tried to use Arduino IDE. It's the same.
I have of course checked my wires multiple times.
Code executes. I have added some prints that are visible on the console.
I really don't understand why it doesn't work.

If someone has an idea...
Or a mean to debug.

**Comment by N0ury** [comment]
Last info: I've tried this example using Adafruit_GFX and Adafruit_ILI9341 and it works fine.
I had just to change pin numbers.
I'm quiet sure now that it is related to TFT_eSPI lib.

**Comment by Bodmer** [comment]
Use setup 71 as a starting point and check your pin allocations so the native fast hardware pins are used.

**Comment by N0ury** [comment]
Are you sure? Setup 71 is for ST7789 not ILI9341
On the package I have received it is clearly an ILI9341.
I have even asked the seller. Response was "yes, ILI9341 as you asked for".

**Comment by Bodmer** [comment]
Use it as a starting point to create your own setup as it is tested, just change the driver.

**Comment by N0ury** [comment]
@Bodmer I've done what you asked for.
I'm very surprised by the result.
Here's what I've done:
- test have been done with Arduino IDE Version 2.2.1 on MacOS
- I've modified User_Setup_Select.h to make it use Setup71_ESP32_S2_ST7789.h
- I've modified Setup71_ESP32_S2_ST7789.h: ST7789_DRIVER => ILI9341_DRIVER
- I've the swaped the pin to make them reflect what was in the file
- Uploaded and run the sketch. It works!!!
So we have now 
#define TFT_SCLK 37
#define TFT_MISO 36
and Espressif doc says:
<img width="641" alt="Capture d’écran 2023-11-05 à 15 46 20" src="https://github.com/Bodmer/TFT_eSPI/assets/4224717/34c42c56-1e0c-4f0e-ad97-39b002a924af">

You must admit with me that it's really strange.

But the fact is that with this conf it works.

**Comment by N0ury** [comment]
Furthermore it works with `SPI_FREQUENCY   27000000` not with `SPI_FREQUENCY   40000000`

**Comment by N0ury** [comment]
And last, it works the same way with PlatformIO

**Comment by Bodmer** [comment]
OK, it looks like either[ setup 71 is wrong](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup71_ESP32_S2_ST7789.h#L11-L12) as the pins are swapped compared to the comments. Maybe there was a specific ESP32 S2 hardware board it was created for as I see the CS line is not used either.

The interface should work with the pinout in the comments (correct) or with those as defined (36 & 37 swapped), so I am not sure what is happening.  Typically ILI9341 boards work up to 80MHz so maybe you have one of the China clone chips I have seen mentioned on the web. I seem to have genuine ILI9341 chips though and have not seen a clone.

Glad to hear you have it working.
